searchcore/src/vespa/searchcore/proton/persistenceengine/i_document_retriever.h

```diff
@@ -47,13 +54,17 @@ class IDocumentRetriever
     virtual void visitDocuments(const LidVector &lids, search::IDocumentVisitor &visitor, ReadConsistency readConsistency) const = 0;
 
     virtual CachedSelect::SP parseSelect(const vespalib::string &selection) const = 0;
+
+    // Convenience to get all fields
+    DocumentUP getDocument(search::DocumentIdT lid, const document::DocumentId & docId) const;
```

Comment:
not sure how much convenience should be in the API, but if there is convenience, consider adding the default implementation of get(Partial)Document as well, to get rid of duplication.

Reply:
Fixed